File: src/core/src/main/java/com/github/malavv/brewru/inventory/Stock.java
package com.github.malavv.brewru.inventory;

import com.github.malavv.brewru.unit.Quantity;

import java.time.LocalDate;

public class Stock {
  public Item item;
  public Quantity quantity;
  public LocalDate boughtOn;
  public String provider;

  public Stock(Quantity quantity, LocalDate boughtOn, String provider) {
    this.quantity = quantity;
    this.boughtOn = boughtOn;
    this.provider = provider;
  }
}

File: src/builder/core/supply/stock.ts
class Stock {
  private quantity: Quantity;
  private boughtOn: Date;
  private provider: String;

  public static fromRaw(raw: any): Stock {
    var s = new Stock();
    s.quantity = raw.quantity;
    s.boughtOn = new Date(raw.boughtOn);
    s.provider = raw.provider;
    return s;
  }
}
